rfcs/2022-04-20-12187-log-namespacing.md

```diff
@@ -0,0 +1,399 @@
+# RFC 12187 - 2022-04-20 - Log Namesapcing
+
+Today, when deserializing data on an event the keys are arbitrary and can potentially collide with data on the root.
+Not only is the loss of data inconvenient, but it prevents us from fully utilizing the power of schemas.
+The event data should be restructured to prevent collisions, and in general make them easier to use.
+
+
+## Proposal
+
+### Changes
+
+Support will be added for different log namespaces in Vector. The default will be a "Legacy" namespace, which keeps
+behavior the same as it was before namespaces were introduced. A new "Vector" namespace will be added with the changes
```

Comment:
Is it worth versioning this `Vector` namespace to allow for future iterations, if needed?

Reply:
Right now the name only appears in the enum variant in the code, and users don't see it. If needed, the name can be changed without any issues. I'd prefer to keep it as is, and in the future if a new version is needed we can choose new names for both.